// @mistralai/mistralai/esm/sdk/connectors.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 2dda252d574b
 */
import { betaConnectorsCallTool } from "../funcs/betaConnectorsCallTool.js";
import { betaConnectorsCreate } from "../funcs/betaConnectorsCreate.js";
import { betaConnectorsDelete } from "../funcs/betaConnectorsDelete.js";
import { betaConnectorsGet } from "../funcs/betaConnectorsGet.js";
import { betaConnectorsGetAuthUrl } from "../funcs/betaConnectorsGetAuthUrl.js";
import { betaConnectorsList } from "../funcs/betaConnectorsList.js";
import { betaConnectorsListTools } from "../funcs/betaConnectorsListTools.js";
import { betaConnectorsUpdate } from "../funcs/betaConnectorsUpdate.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
export class Connectors extends ClientSDK {
    /**
     * Create a new connector.
     *
     * @remarks
     * Create a new MCP connector. You can customize its visibility, url and auth type.
     */
    async create(request, options) {
        return unwrapAsync(betaConnectorsCreate(this, request, options));
    }
    /**
     * List all connectors.
     *
     * @remarks
     * List all your custom connectors with keyset pagination and filters.
     */
    async list(request, options) {
        return unwrapAsync(betaConnectorsList(this, request, options));
    }
    /**
     * Get the auth URL for a connector.
     *
     * @remarks
     * Get the OAuth2 authorization URL for a connector to initiate user authentication.
     */
    async getAuthUrl(request, options) {
        return unwrapAsync(betaConnectorsGetAuthUrl(this, request, options));
    }
    /**
     * Call Connector Tool
     *
     * @remarks
     * Call a tool on an MCP connector.
     */
    async callTool(request, options) {
        return unwrapAsync(betaConnectorsCallTool(this, request, options));
    }
    /**
     * List tools for a connector.
     *
     * @remarks
     * List all tools available for an MCP connector.
     */
    async listTools(request, options) {
        return unwrapAsync(betaConnectorsListTools(this, request, options));
    }
    /**
     * Get a connector.
     *
     * @remarks
     * Get a connector by its ID or name.
     */
    async get(request, options) {
        return unwrapAsync(betaConnectorsGet(this, request, options));
    }
    /**
     * Update a connector.
     *
     * @remarks
     * Update a connector by its ID.
     */
    async update(request, options) {
        return unwrapAsync(betaConnectorsUpdate(this, request, options));
    }
    /**
     * Delete a connector.
     *
     * @remarks
     * Delete a connector by its ID.
     */
    async delete(request, options) {
        return unwrapAsync(betaConnectorsDelete(this, request, options));
    }
}
