// @mistralai/mistralai/esm/models/components/internalmetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5bb58674fbba
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { MirrorConfig$inboundSchema, MirrorConfig$outboundSchema, } from "./mirrorconfig.js";
/** @internal */
export const InternalMetadataAgentVersion$inboundSchema = smartUnion([z.string(), z.int()]);
/** @internal */
export const InternalMetadataAgentVersion$outboundSchema = smartUnion([z.string(), z.int()]);
export function internalMetadataAgentVersionToJSON(internalMetadataAgentVersion) {
    return JSON.stringify(InternalMetadataAgentVersion$outboundSchema.parse(internalMetadataAgentVersion));
}
export function internalMetadataAgentVersionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InternalMetadataAgentVersion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InternalMetadataAgentVersion' from JSON`);
}
/** @internal */
export const InternalMetadata$inboundSchema = z.object({
    agent_id: z.nullable(z.string()).optional(),
    agent_version: z.nullable(smartUnion([z.string(), z.int()])).optional(),
    mirror_config: z.nullable(MirrorConfig$inboundSchema).optional(),
    science_config: z.nullable(z.record(z.string(), z.any())).optional(),
}).catchall(z.any()).transform((v) => {
    return remap$(v, {
        "agent_id": "agentId",
        "agent_version": "agentVersion",
        "mirror_config": "mirrorConfig",
        "science_config": "scienceConfig",
    });
});
/** @internal */
export const InternalMetadata$outboundSchema = z.object({
    agentId: z.nullable(z.string()).optional(),
    agentVersion: z.nullable(smartUnion([z.string(), z.int()])).optional(),
    mirrorConfig: z.nullable(MirrorConfig$outboundSchema).optional(),
    scienceConfig: z.nullable(z.record(z.string(), z.any())).optional(),
}).catchall(z.any()).transform((v) => {
    return {
        ...remap$(v, {
            agentId: "agent_id",
            agentVersion: "agent_version",
            mirrorConfig: "mirror_config",
            scienceConfig: "science_config",
        }),
    };
});
export function internalMetadataToJSON(internalMetadata) {
    return JSON.stringify(InternalMetadata$outboundSchema.parse(internalMetadata));
}
export function internalMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InternalMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InternalMetadata' from JSON`);
}
