// @mistralai/mistralai/esm/models/components/mirrorsource.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3bd93f177873
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const MirrorSource$inboundSchema = z
    .object({
    app: z.string(),
    version: z.nullable(z.string()).optional(),
    root_model_name: z.string(),
    mirror_model_name: z.string(),
    experiment_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "root_model_name": "rootModelName",
        "mirror_model_name": "mirrorModelName",
        "experiment_id": "experimentId",
    });
});
/** @internal */
export const MirrorSource$outboundSchema = z.object({
    app: z.string(),
    version: z.nullable(z.string()).optional(),
    rootModelName: z.string(),
    mirrorModelName: z.string(),
    experimentId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        rootModelName: "root_model_name",
        mirrorModelName: "mirror_model_name",
        experimentId: "experiment_id",
    });
});
export function mirrorSourceToJSON(mirrorSource) {
    return JSON.stringify(MirrorSource$outboundSchema.parse(mirrorSource));
}
export function mirrorSourceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MirrorSource$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MirrorSource' from JSON`);
}
