// @mistralai/mistralai/esm/models/components/mirrorconfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: c8c262460c15
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { MirrorSource$inboundSchema, MirrorSource$outboundSchema, } from "./mirrorsource.js";
/** @internal */
export const MirrorConfig$inboundSchema = z
    .object({
    mirror_source: MirrorSource$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "mirror_source": "mirrorSource",
    });
});
/** @internal */
export const MirrorConfig$outboundSchema = z.object({
    mirrorSource: MirrorSource$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        mirrorSource: "mirror_source",
    });
});
export function mirrorConfigToJSON(mirrorConfig) {
    return JSON.stringify(MirrorConfig$outboundSchema.parse(mirrorConfig));
}
export function mirrorConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MirrorConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MirrorConfig' from JSON`);
}
